src/msg.jl

```diff
@@ -143,14 +143,18 @@ end
 # adds notation message.field
 
 function Base.getproperty(s::Message, p::Symbol)
-  hasfield(typeof(s), p) || return nothing
+  hasfield(typeof(s), p) || error("message $(typeof(s)) has no field $(p)")
   getfield(s, p)
 end
 
-function Base.setproperty!(s::Message, p::Symbol, v)
-  hasfield(typeof(s), p) || return s
-  ftype = fieldtype(typeof(s), p)
-  setfield!(s, p, convert(ftype, v))
+function Base.setproperty!(s::Message, p::Symbol, v; ignore_missingfields=false)
```

Comment:
If we split out `setproperty_nothrow!()` into a separate function, then `setproperty!()` could also be the default implementation...

Reply:
Fair enough. Implemented.